modules/homebrew.nix

```diff
@@ -1,219 +1,751 @@
+      };
+      dictionarydir = mkNullOrStrOption {
+        description = ''
+          Target location for Dictionaries.
+
+          Homebrew's default is <filename class='directory'>~/Library/Dictionaries</filename>.
+        '';
+      };
+      fontdir = mkNullOrStrOption {
+        description = ''
+          Target location for Fonts.
+
+          Homebrew's default is <filename class='directory'>~/Library/Fonts</filename>.
+        '';
+      };
+      servicedir = mkNullOrStrOption {
+        description = ''
+          Target location for Services.
+
+          Homebrew's default is <filename class='directory'>~/Library/Services</filename>.
+        '';
+      };
+      input_methoddir = mkNullOrStrOption {
+        description = ''
+          Target location for Input Methods.
+
+          Homebrew's default is <filename class='directory'>~/Library/Input Methods</filename>.
+        '';
+      };
+      internet_plugindir = mkNullOrStrOption {
+        description = ''
+          Target location for Internet Plugins.
+
+          Homebrew's default is <filename class='directory'>~/Library/Internet Plug-Ins</filename>.
+        '';
+      };
+      audio_unit_plugindir = mkNullOrStrOption {
+        description = ''
+          Target location for Audio Unit Plugins.
+
+          Homebrew's default is
+          <filename class='directory'>~/Library/Audio/Plug-Ins/Components</filename>.
+        '';
+      };
+      vst_plugindir = mkNullOrStrOption {
+        description = ''
+          Target location for VST Plugins.
+
+          Homebrew's default is <filename class='directory'>~/Library/Audio/Plug-Ins/VST</filename>.
+        '';
+      };
+      vst3_plugindir = mkNullOrStrOption {
+        description = ''
+          Target location for VST3 Plugins.
+
+          Homebrew's default is <filename class='directory'>~/Library/Audio/Plug-Ins/VST3</filename>.
+        '';
+      };
+      screen_saverdir = mkNullOrStrOption {
+        description = ''
+          Target location for Screen Savers.
+
+          Homebrew's default is <filename class='directory'>~/Library/Screen Savers</filename>.
+        '';
+      };
+      language = mkNullOrStrOption {
+        description = ''
+          Comma-separated list of language codes to prefer for cask installation. The first matching
+          language is used, otherwise it reverts to the cask’s default language. The default value
+          is the language of your system.
+        '';
+        example = "zh-TW";
+      };
+      require_sha = mkNullOrBoolOption {
+        description = ''
+          Whether to require cask(s) to have a checksum.
+
+          Homebrew's default is <literal>false</literal>.
+        '';
+      };
+      no_quarantine = mkNullOrBoolOption {
+        description = "Whether to disable quarantining of downloads.";
+      };
+      no_binaries = mkNullOrBoolOption {
+        description = "Whether to disable linking of helper executables.";
+      };
```

Comment:
```suggestion
      quarantine = mkNullOrBoolOption {
        description = "Whether to enable quarantining of downloads.";
      };
      binaries = mkNullOrBoolOption {
        description = "Whether to enable linking of helper executables.";
      };
```

Reply:
All the option names here for the `homebrew.caskArgs` are exact matches for valid arguments for the `cask_args` entry of a Brewfile (and the `args` argument for `casks` entries). For example, enabling those two options will create the following entry in the Brewfile:

```ruby
cask_args no_binaries: true, no_quarantine: false
# or
casks "google-chrome", args: { no_binaries: true, no_quarantine: false }
```

I think it's best to have the option names here match what ends up in the Brewfile. Additionally I don't like the idea of having to add special cases in the code to process these two options.

Reply:
That's unfortunate.  There's both `--[no-]{binaries,quarantine}` options so I figured both ways might be valid. The inverted boolean flag is an unnecessarily confusing anti-pattern.

Reply:
So I will confess that I'm not completely sure how `brew bundle` is doing things under the hood here, but I'm pretty sure what it's doing is pretty naive. My guess is that, for binary arguments provided to `cask_args`, if they are set to `true`, the corresponding flag is added to the `brew install --cask` command that it runs, and if set they are to `false` it just leave off those flags.

Reply:
I'm also inferring this from the Homebrew manpages,

> HOMEBREW_CASK_OPTS 
> Append these options to all cask commands. All --*dir options, --language, --require-sha, --no-quarantine and --no-binaries are supported.

and the fact that, by default, quarantining and linking of binaries is enabled.

Reply:
This is up to you again. It's a bit of a tug of war between simpler UX on one hand and simple implementation on the other.

Reply:
> It's a bit of a tug of war between simpler UX on one hand and simple implementation on the other.

Totally.

Thanks, for the feedback. In this case I'd like to leave these as is for the reasons I mentioned above related to uncertainty about how `brew bundle` works under the hood, and therefore wanting the line in the Brewfile to always include the `no_` prefixed version of the options, and not wanting to add custom processing in the submodule to rename them while still using the `no_` prefix in the generated brewfileLine for the submodule.